src/main/java/frc/robot/commands/AutomaticShootCommand.java

```diff
@@ -0,0 +1,43 @@
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import edu.wpi.first.wpilibj2.command.InstantCommand;
+import edu.wpi.first.wpilibj2.command.button.Trigger;
+import frc.robot.subsystems.ShooterSubsystem;
+
+public class AutomaticShootCommand extends CommandBase {
+    private final ShooterSubsystem shooterSubsystem;
+    private final Trigger ballSensorTrigger;
+    private final double targetVel;
+    private int ballsLeftToShoot;
+
+    public AutomaticShootCommand(double targetVel, int ballsLeft, ShooterSubsystem shooterSubsystem) {
+        this.targetVel = targetVel;
+        this.ballsLeftToShoot = ballsLeft;
+        this.shooterSubsystem = shooterSubsystem;
+        addRequirements(shooterSubsystem);
+        ballSensorTrigger = new Trigger(this.shooterSubsystem::isBallReady);
+        ballSensorTrigger.whenActive(
+                new InstantCommand(
+                        () -> {
+                            ballsLeftToShoot--;
+                            // logger.info("Total Balls Shot: " + totalShoot);
+                        }));
```

Comment:
Did you test this? Is it possible that this gets triggered too many times and a ball is in the robot but unable to be shot?

Reply:
We just tested this and it has been working very well. We are going to test it in autonomous in a bit.